### PR TITLE
Severenity iOS VIPER migration

### PR DESCRIPTION
**Pull request by YuraYasinskyy**
Severenity for iOS now works on VIPER architecture. Stable chat implementation. A lot of bug fixes and behavior improvements.

### CODE REVIEW

File: iOS/Severenity/Interactors/ChatInteractor.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    
+    weak var delegate: ChatInteractorDelegate?
+    
+    // MARK: - Init
+    
+    override init() {
+        super.init()
+        WireFrame.sharedInstance.viperInteractors["ChatInteractor"] = self
+    }
+
```
Comment (OlegNovosad) [inline iOS/Severenity/Interactors/ChatInteractor.swift]: Don't leave so much space between methods. One empty line is enough

Comment (YuraYasinskyy) [inline iOS/Severenity/Interactors/ChatInteractor.swift]: Fixed.

File: iOS/Severenity/Views/Chat/ChatViewController.swift [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        let message = messages[(indexPath as NSIndexPath).row]
+        
+        guard let senderName = message["senderName"] as? String, let timestamp = message["timestamp"] as? String,
+            let messageText = message["text"] as? String, let senderFbId = message["senderId"] as? String else {
+            print("Cannot create chat message cell")
+            cell = UITableViewCell()
+            return cell
+        }
+        
+        if message["senderId"] as! String != (FBSDKAccessToken.current().userID)! {
```
Comment (OlegNovosad) [inline iOS/Severenity/Views/Chat/ChatViewController.swift]: Use **senderFbId** from guard statement.

File: iOS/Severenity/Views/MapViewController.swift [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            mapView.delegate = self
+            view = mapView
+            mapView.isMyLocationEnabled = false
+        }
+    }
+    
+    /**- Calling this method simply adjust Google Map to see all markers */
+    private func showAllMarkersOnMap() {
+        let firstPlace = markers.first?.value.position
+        var bounds = GMSCoordinateBounds.init(coordinate: firstPlace!, coordinate: firstPlace!)
```
Comment (OlegNovosad) [inline iOS/Severenity/Views/MapViewController.swift]: Remove force unwrapping for the first place. Use guard instead.

File: iOS/Severenity/Views/MapViewController.swift [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        return maskedImage
+    }
+    
+    func addNewPlayerToMap(with image: UIImage, and coordinates: CLLocationCoordinate2D, and info: Dictionary<String,String>) {
+        guard let userId = info["id"], let userName = info["name"] else {
+            print("Cannot add player marker to map with recieved info")
+            return
+        }
+        if markers[userId] != nil {
+            var customImage = image.roundedImageWithBorder(with: 5, and: #colorLiteral(red: 0.5176470588, green: 0.3411764706, blue: 0.6, alpha: 1))
```
Comment (OlegNovosad) [inline iOS/Severenity/Views/MapViewController.swift]: custom image is used in two blocks

File: iOS/Severenity/Interactors/ChatInteractor.swift [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    func sendChatMessage(with text: String) {
+        FacebookService.sharedInstance.getFBProfileInfo(with: (FBSDKAccessToken.current().userID)!, and: { (info) in
+            let messageJSON = ["senderName":info["name"] ?? "",
+                               "senderId":(FBSDKAccessToken.current().userID)!,
+                               "text":text,
+                               "timestamp":self.currentTimeStamp] as [String:Any]
+            SocketService.sharedInstance.sendMessageToServer(with: messageJSON)
+            
+        })
+        print("ChatInteractor was called from ChatPresenter to send message")
```
Comment (YuraYasinskyy) [inline iOS/Severenity/Interactors/ChatInteractor.swift]: Fixed.

File: iOS/Severenity/Interactors/MapInteractor.swift [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        super.init()
+        WireFrame.sharedInstance.viperInteractors["MapInteractor"] = self
+    }
+    
+    // MARK: - Other events
+    
+    func profileListViewEvent(with data: AnyObject) {
+        print("MapInteractor event happened from ProfileListView with data: \(data)")
+        if let recievedData = data as? Dictionary<String,Any> {
+            delegate?.displayPlace(with: recievedData as Dictionary<String, Any>)
```
Comment (YuraYasinskyy) [inline iOS/Severenity/Interactors/MapInteractor.swift]: Actually no. Fixed.

File: iOS/Severenity/Services/FacebookService.swift [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            
+            if let image = response.result.value {
+                print("Facebook profile picture downloaded")
+                completion(image)
+            }
+        }
+    }
+    
+    func getFBProfileInfo(with fbUserID: String, and completion: @escaping (_ info: Dictionary<String,String>) -> Void){
+        FBSDKGraphRequest(graphPath: fbUserID, parameters: ["fields": "id, name, first_name, last_name, relationship_status"]).start(completionHandler: { (connection, result, error) -> Void in
```
Comment (YuraYasinskyy) [inline iOS/Severenity/Services/FacebookService.swift]: Added.

File: iOS/Severenity/Services/FacebookService.swift [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            if (error == nil){
```
Comment (YuraYasinskyy) [inline iOS/Severenity/Services/FacebookService.swift]: Fixed.

File: iOS/Severenity/Services/PlacesService.swift [hunk 9]
```diff
@@ -27,6 +25,8 @@ class LocationsServerManager: NSObject {
         }
```
Comment (YuraYasinskyy) [inline iOS/Severenity/Services/PlacesService.swift]: Actually I catch. I believe it's just GitHub cropping some lines.

File: iOS/Severenity/Services/PlacesService.swift [hunk 10]
```diff
@@ -46,17 +46,17 @@ class LocationsServerManager: NSObject {
...
     }
     
-    func checkIfRealmIsEmpty() -> Bool {
+    private func checkIfRealmIsEmpty() -> Bool {
         let realmReadQuery = self.realm?.objects(RealmPlace.self)
```
Comment (YuraYasinskyy) [inline iOS/Severenity/Services/PlacesService.swift]: Ok, fixed it.

File: iOS/Severenity/Views/Chat/ChatViewController.swift (hunk 2 above)
Comment (YuraYasinskyy) [inline iOS/Severenity/Views/Chat/ChatViewController.swift]: Fixed.

File: iOS/Severenity/Views/MapViewController.swift (hunk 3 above)
Comment (YuraYasinskyy) [inline iOS/Severenity/Views/MapViewController.swift]: Fixed.

File: iOS/Severenity/Views/MapViewController.swift (hunk 4 above)
Comment (YuraYasinskyy) [inline iOS/Severenity/Views/MapViewController.swift]: Refactored it.

**Pull request merged**